modules/apps/dynamic-data-lists/dynamic-data-lists-service/src/main/java/com/liferay/dynamic/data/lists/internal/change/tracking/spi/reference/DDLRecordTableReferenceDefinition.java

```diff
@@ -51,6 +52,22 @@ public void defineChildTableReferences(
 						AssetEntryTable.INSTANCE.classNameId)
 				)
 			)
+		).referenceInnerJoin(
+			fromStep -> fromStep.from(
+				DDMFieldTable.INSTANCE
+			).innerJoinON(
+				DDLRecordTable.INSTANCE,
```

Comment:
Why are you doing an inner join by companyId here?

Reply:
Because SF requires `DDLRecord` to be the first in the joins, and we can only join the `DDLRecord` and `DDMField` on companyId. This is what the query should look like, but SF doesn't allow it...

```
select * from DDMField 
inner join DDLRecordVersion on DDLRecordVersion.DDMStorageId = DDMField.storageId
inner Join DDLRecord on DDLRecord.recordId = DDLRecordVersion.recordId
where DDLRecord.ctCollectionId = ?
```